sharedresource/v1alpha1/0000_10_sharedresource.crd.yaml

```
@@ -0,0 +1,112 @@
+                  description: Description is a user readable explanation of what the backing resource provides.
+                  type: string
+            status:
+              description: ShareStatus defines the observed state of Share
+              type: object
+              properties:
+                conditions:
+                  description: Conditions are the set of k8s Condition instances provided by the associated controller for Shares.
+                  type: array
+                  items:
+                    description: "Condition contains details for one aspect of the current state of this API Resource. --- This struct is intended for direct use as an array at the field path .status.conditions.  For example, type FooStatus struct{     // Represents the observations of a foo's current state.     // Known .status.conditions.type are: \"Available\", \"Progressing\", and \"Degraded\"     // +patchMergeKey=type     // +patchStrategy=merge     // +listType=map     // +listMapKey=type     Conditions []metav1.Condition `json:\"conditions,omitempty\" patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"` \n     // other fields }"
+                    type: object
+                    required:
+                      - lastTransitionTime
+                      - message
+                      - reason
+                      - status
+                      - type
+                    properties:
+                      lastTransitionTime:
+                        description: lastTransitionTime is the last time the condition transitioned from one status to another. This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
+                        type: string
+                        format: date-time
+                      message:
+                        description: message is a human readable message indicating details about the transition. This may be an empty string.
+                        type: string
+                        maxLength: 32768
+                      observedGeneration:
+                        description: observedGeneration represents the .metadata.generation that the condition was set based upon. For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date with respect to the current state of the instance.
+                        type: integer
+                        format: int64
+                        minimum: 0
+                      reason:
+                        description: reason contains a programmatic identifier indicating the reason for the condition's last transition. Producers of specific condition types may define expected values and meanings for this field, and whether the values are considered a guaranteed API. The value should be a CamelCase string. This field may not be empty.
```

```suggestion
                        description: `reason` contains a programmatic identifier indicating the reason for the condition's last transition. Producers of specific condition types can define expected values and meanings for this field, and whether the values are considered a guaranteed API. The value should be a CamelCase string. This field cannot not be empty.
```
For localization, please replace instances of "may" in this PR with "can" or "might." 

From our style guide:
Use "might" to indicate possibility. To indicate ability, use "can".
In most cases, use "can" or "might", not "may".

---

Do we want to start putting field names like `reason` inside backticks? This helps readers identify it as a field name and helps localization teams avoid translating it.

---

Same here: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1398-L1444